bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NhcSystemInfo2.java

```diff
@@ -42,23 +43,23 @@
     ArrayList<NhcSwVersion> swVersions = new ArrayList<>();
 
     /**
-     * @return the lastConfig
+     * @return the NhcVersion
      */
     public String getNhcVersion() {
-        if (swVersions.size() > 0) {
-            return swVersions.get(0).nhcVersion;
-        } else {
+        try {
```

Comment:
```suggestion
        swVersions.stream().map(p -> p.nhcVersion).filter(v -> !v.isEmpty()).findFirst().orElse("");
```

Reply:
Thanks for this. I am still learning to use these streams properly.